unifi-poller prometheus exporter module no longer functional in NixOS

### Describe the bug
The upgrade of unifi-poller in #199854 has changed the executable name, and the related Prometheus exporter configuration by NixOS module no longer works.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Enable unifi-poller in NixOS:
```
  services.prometheus.exporters.unifi-poller = {
    enable = true;
    controllers = [{
      url = "https://${unifiHost}";
      user = "readonly";
      pass = /etc/nixos/unifi.pass;
    }];
  };
```
2. Observe that the systemd process can't start...
```
$ sudo systemctl status prometheus-unifi-poller-exporter.service
× prometheus-unifi-poller-exporter.service
     Loaded: loaded (/etc/systemd/system/prometheus-unifi-poller-exporter.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Tue 2022-11-15 15:28:37 MST; 4min 26s ago
   Duration: 120ms
    Process: 58603 ExecStart=/nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7/bin/unifi-poller --config />
   Main PID: 58603 (code=exited, status=203/EXEC)
         IP: 0B in, 0B out
        CPU: 118ms

$ journalctl -u prometheus-unifi-poller-exporter.service
...
Nov 15 15:28:37 anxi systemd[1]: Started prometheus-unifi-poller-exporter.service.
Nov 15 15:28:37 anxi systemd[58603]: prometheus-unifi-poller-exporter.service: Failed to locate executable /nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7/bin/unifi-poller: No such file or directory
Nov 15 15:28:37 anxi systemd[58603]: prometheus-unifi-poller-exporter.service: Failed at step EXEC spawning /nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7/bin/unifi-poller: No such file or directory
Nov 15 15:28:37 anxi systemd[1]: prometheus-unifi-poller-exporter.service: Main process exited, code=exited, status=203/EXEC
...
```


It appears that the upgrade from 2.1.3 has changed the binary name from unifi-poller to unpoller, and the NixOS module has not had an associated change to support it.

```
$ find /nix/store/dy9fvwy9jrfqphwj1bfvbvra3xfryacx-unifi-poller-2.1.3
/nix/store/dy9fvwy9jrfqphwj1bfvbvra3xfryacx-unifi-poller-2.1.3
/nix/store/dy9fvwy9jrfqphwj1bfvbvra3xfryacx-unifi-poller-2.1.3/bin
/nix/store/dy9fvwy9jrfqphwj1bfvbvra3xfryacx-unifi-poller-2.1.3/bin/unifi-poller

$ find /nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7
/nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7
/nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7/bin
/nix/store/xd468v1nibc91mgpflgzq8pwd6ylqxvl-unifi-poller-2.1.7/bin/unpoller
```

### Expected behavior
When the NixOS module `unifi-poller` is enabled, the service it creates should start up.

### Notify maintainers

@fpletz - merged #199854
@ryantm - bot created #199854

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.78, NixOS, 22.11 (Raccoon), 22.11pre426637.013fcdd1068`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"mfenniak, nixos"`
 - channels(mfenniak): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Thanks, new minor of the exporter also segfaults when running the test:

```
unifi_poller # [    6.758446] systemd[1]: Started prometheus-unifi-poller-exporter.service.
unifi_poller # [    6.778981] unpoller[906]: 2022/11/16 00:53:53 [INFO] Loading Configuration File: /nix/store/wz5ihl4zwjww28n5nbiaxbkafxsns7ni-prometheus-unifi-poller-exporter.json
unifi_poller # [    6.781814] unpoller[906]: 2022/11/16 00:53:53 [INFO] UniFi Poller v Starting Up! PID: 906
unifi_poller # [    6.783904] unpoller[906]: 2022/11/16 00:53:53 [ERROR] Controller 1 of 1 Auth or Connection Error, retrying: unifi controller: making request: Get "https://unifi:8443/": dial tcp: lookup unifi: Temporary failure in name resolution
unifi_poller # [    6.785718] unpoller[906]: 2022/11/16 00:53:53 [INFO] Prometheus exported at http://0.0.0.0:9130/ - namespace: unpoller
unifi_poller # [    6.786505] unpoller[906]: 2022/11/16 00:53:53 [INFO] Internal web server disabled!
unifi_poller # [    6.789209] unpoller[906]: panic: runtime error: invalid memory address or nil pointer dereference
unifi_poller # [    6.789873] unpoller[906]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x9 pc=0xb43e44]
unifi_poller # [    6.790416] unpoller[906]: goroutine 12 [running]:
unifi_poller # [    6.790745] unpoller[906]: github.com/unpoller/datadogunifi.(*DatadogUnifi).Run(0x0?, {0xeddae8?, 0xc00005fde0?})
unifi_poller # [    6.791323] unpoller[906]:    github.com/unpoller/datadogunifi@v0.0.0-20221109161507-2c4fbce23b95/datadog.go:194 +0x44
unifi_poller # [    6.792329] unpoller[906]: github.com/unpoller/poller.(*UnifiPoller).runOutputMethods.func1(0x0?)
unifi_poller # [    6.792948] unpoller[906]:    github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:85 +0x31
unifi_poller # [    6.794704] unpoller[906]: created by github.com/unpoller/poller.(*UnifiPoller).runOutputMethods
unifi_poller # [    6.795215] unpoller[906]:    github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:84 +0xd0
unifi_poller # [    6.795877] systemd[1]: prometheus-unifi-poller-exporter.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
unifi_poller # [    6.796580] systemd[1]: prometheus-unifi-poller-exporter.service: Failed with result 'exit-code'.
unifi_poller # [    6.797133] systemd[1]: prometheus-unifi-poller-exporter.service: Consumed 19ms CPU time, received 0B IP traffic, sent 248B IP traffic.
unifi_poller # [    7.151815] systemd[1]: prometheus-unifi-poller-exporter.service: Scheduled restart job, restart counter is at 4.
unifi_poller # [    7.154627] systemd[1]: Stopped prometheus-unifi-poller-exporter.service.
unifi_poller # [    7.157406] systemd[1]: prometheus-unifi-poller-exporter.service: Consumed 19ms CPU time, received 0B IP traffic, sent 248B IP traffic.
```

Going to revert to the old version.

--- a/pkgs/servers/monitoring/unifi-poller/default.nix
+++ b/pkgs/servers/monitoring/unifi-poller/default.nix
@@ -2,16 +2,16 @@
 
 buildGoModule rec {
   pname = "unifi-poller";
-  version = "2.1.7";
+  version = "2.1.3";
 
   src = fetchFromGitHub {
     owner = "unifi-poller";
     repo = "unifi-poller";
     rev = "v${version}";
-    sha256 = "sha256-79aBqYS8B+gIqmhu/UIA7g+FvTS/68qsK7qi5MGH10k=";
+    sha256 = "sha256-xh9s1xAhIeEmeDprl7iPdE6pxmxZjzgMvilobiIoJp0=";
   };
 
-  vendorSha256 = "sha256-WVYQ3cZOO+EyJRTFcMjziDHwqqinm1IvxvSLuHTaqT8=";
+  vendorSha256 = "sha256-HoYgBKTl9HIMVzzzNYtRrfmqb7HCpPHVPeR4gUXneWk=";
 
   ldflags = [
     "-w" "-s"

